server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -602,64 +631,39 @@ private static SnapshotsInRepo sortSnapshots(
         return new SnapshotsInRepo(resultSet, snapshotInfos.size(), allSnapshots.size() - resultSet.size());
     }
 
-    private static Predicate<SnapshotInfo> buildFromSortValuePredicate(
+    private static Predicate<SnapshotInfo> buildAfterPredicate(
         GetSnapshotsRequest.SortBy sortBy,
-        String after,
-        SortOrder order,
-        @Nullable String snapshotName,
-        @Nullable String repoName
+        GetSnapshotsRequest.After after,
+        SortOrder order
     ) {
-        final Predicate<SnapshotInfo> isAfter;
+        final String snapshotName = after.snapshotName();
+        final String repoName = after.repoName();
+        final String value = after.value();
         switch (sortBy) {
             case START_TIME:
-                isAfter = filterByLongOffset(SnapshotInfo::startTime, Long.parseLong(after), snapshotName, repoName, order);
-                break;
+                return filterByLongOffset(SnapshotInfo::startTime, Long.parseLong(value), snapshotName, repoName, order);
             case NAME:
-                if (snapshotName == null) {
-                    assert repoName == null : "no snapshot name given but saw repo name [" + repoName + "]";
-                    isAfter = order == SortOrder.ASC
-                        ? snapshotInfo -> after.compareTo(snapshotInfo.snapshotId().getName()) <= 0
-                        : snapshotInfo -> after.compareTo(snapshotInfo.snapshotId().getName()) >= 0;
-                } else {
-                    isAfter = order == SortOrder.ASC
-                        ? (info -> compareName(snapshotName, repoName, info) < 0)
-                        : (info -> compareName(snapshotName, repoName, info) > 0);
-                }
-                break;
+                // TODO: cover via pre-flight predicate
```

Comment:
I am not sure you can unless we change how the count works? Though I guess we could possibly avoid getting the snapshot info for the "before" snapshots if they are not needed.